woocommerce/woocommerce-android
# [Login] Site discovery post-login, part 2

Closes: #7215 

### Description
This PR builds on top of the previous PR #7223, it adds the following functionality:
1. Jetpack required state
2. Jetpack installation flow from the Jetpack error state.
3. Error state for non-wordpress sites.
4. Handle logout from the error states.
5. Display inline error for invalid address (this is commit b6b48a863890d06ebad161f46ee572ebe5a89a9c, I recommend reviewing the commit separately to make the review easier).

### Testing instructions
1. Open the app then sign in with a WordPress.com account that's not connected to any site.
2. Click on "Enter a site address"
3. Type "google.com" then click on "Continue"
4. Confirm in the logs the tracking event: `site_picker_site_discovery, Properties: {"has_wordpress":false,"is_wpcom":false,"has_valid_jetpack":false}`
5. Click on "Try another store"
6. Enter the address of a site that doesn't have Jetpack then click on "Continue"
7. Confirm in the logs the tracking event: `site_picker_site_discovery, Properties: {"has_wordpress":true,"is_wpcom":false,"has_valid_jetpack":false}`
8. Click on "Install Jetpack"
9. Complete the installation in the WebView. (Confirm that you are automatically logged to the correct WordPress.com account)
10. Confirm the app continues the login automatically after the installation (assuming WooCommerce was installed).

For 10, you can also test with a non-Woo site like the video below.

### Images/gif
https://user-images.githubusercontent.com/1657201/185420795-b52c8e6d-6efd-413b-8eff-12ae3e4b76a7.mov
